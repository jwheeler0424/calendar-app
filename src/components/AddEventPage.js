import React from 'react';

export default () => (
    <div>Add Event Page</div>
);