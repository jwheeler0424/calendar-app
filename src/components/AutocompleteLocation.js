import React from 'react';

const AutocompleteLocation = () => {
    return (
        <input type="text" name="autocomplete-location" placeholder="Location" title="Event Location" />
    );
};

export default AutocompleteLocation;