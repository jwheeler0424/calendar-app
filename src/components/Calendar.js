import React from 'react';
import moment from 'moment';

const currentDate = moment();

const getCalendarDays = (month) => {
    const daysInMonth = month.daysInMonth();
    const monthFirstDay = moment(month).startOf('month');
    const monthFirstDayDOW = monthFirstDay.day();
    const monthLastDay = moment(month).endOf('month');
    const monthLastDayDOW = monthLastDay.day();
    const prevMonthLastDay = moment(month).subtract(1, 'months').endOf('month');
    const nextMonthFirstDay = moment(month).add(1, 'months').startOf('month');
    const calendarDays = [];

    for(let i=prevMonthLastDay.date() - monthFirstDayDOW + 1; i<=prevMonthLastDay.date(); i++) {
        calendarDays.push({
            type: 'pre',
            date: i
        });
    }
    for(let i=monthFirstDay.date(); i<=monthLastDay.date(); i++) {
        calendarDays.push({
            type: 'current',
            date: i
        });
    }
    for(let i=nextMonthFirstDay.date(); i<=(nextMonthFirstDay.date() + (6-monthLastDayDOW) - 1); i++) {
        calendarDays.push({
            type: 'next',
            date: i
        });
    }

    return calendarDays
}

const monthDays = getCalendarDays(currentDate);
console.log(monthDays)
const Calendar = () => (
    <div>
        <h1>Calendar App</h1>
        <h2>{currentDate.format('MMMM')}</h2>
        {monthDays.map((day) => (`${day.date} `))}
    </div>
);

export default Calendar;