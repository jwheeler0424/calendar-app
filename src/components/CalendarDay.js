import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import { setCurrentDate, setActiveView, setLastView } from '../actions/views';
import { setStartDate, setEndDate } from '../actions/filters';

export const CalendarDay = (props) => {
    const selectDate = (date) => {
        const currentDate = moment().set({'date': date.date(), 'month': date.month(), 'year': date.year()});
        const startDate = moment(date.valueOf());
        const endDate = moment(date.valueOf()).endOf('day');
        props.setCurrentDate(currentDate);
        props.setStartDate(startDate);
        props.setEndDate(endDate);
        props.setLastView('');
        props.setActiveView('list');
    };
    
    return (
        <div 
            className={props.type==='current' ? "calendar-day" : "calendar-day pre-next"}
            title={props.date.format("MMM Do")}
            onClick={() => selectDate(props.date)}
            key={props.date}
            date={props.date}
        >
            <div className={props.date.format('MMDDYY') === moment().format('MMDDYY') ? 'content current-day' : 'content'}>
                <div className="day-num">{props.date.format("D")}</div>
            </div>                
        </div>
    );
}

const mapStateToProps = (state) => ({
    views: state.views
});

const mapDispatchToProps = (dispatch) => ({
    setActiveView: (activeView) => dispatch(setActiveView(activeView)),
    setCurrentDate: (currentDate) => dispatch(setCurrentDate(currentDate)),
    setLastView: (lastView) => dispatch(setLastView(lastView)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setStartDate: (startDate) => dispatch(setStartDate(startDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(CalendarDay);