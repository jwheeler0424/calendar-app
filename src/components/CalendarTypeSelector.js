import React from 'react';
import { connect } from 'react-redux';
import {
    setActiveDaily,
    setActiveWeekly,
    setActiveMonthly,
    setActiveYearly
} from '../actions/views';

export const CalendarTypeSelector = (props) => {
    return (
        <div className="calendar-selector__wrapper">
            <button 
                className={
                    props.views.activeCalendar === 'daily' ? (
                        "calendar-selector__active"
                    ) : (
                        "calendar-selector__selector"
                    )
                }
                onClick={props.setActiveDaily}
            >Daily</button>
            <button 
                className={
                    props.views.activeCalendar === 'weekly' ? (
                        "calendar-selector__active"
                    ) : (
                        "calendar-selector__selector"
                    )
                }
                onClick={props.setActiveWeekly}
            >Weekly</button>
            <button 
                className={
                    props.views.activeCalendar === 'monthly' ? (
                        "calendar-selector__active"
                    ) : (
                        "calendar-selector__selector"
                    )
                }
                onClick={props.setActiveMonthly}
            >Monthly</button>
            <button 
                className={
                    props.views.activeCalendar === 'yearly' ? (
                        "calendar-selector__active"
                    ) : (
                        "calendar-selector__selector"
                    )
                }
                onClick={props.setActiveYearly}
            >Yearly</button>
        </div>
    );
}

const mapStateToProps = (state) => ({
    views: state.views
});

const mapDispatchToProps = (dispatch) => ({
    setActiveDaily: () => dispatch(setActiveDaily()),
    setActiveWeekly: () => dispatch(setActiveWeekly()),
    setActiveMonthly: () => dispatch(setActiveMonthly()),
    setActiveYearly: () => dispatch(setActiveYearly())
});

export default connect(mapStateToProps, mapDispatchToProps)(CalendarTypeSelector);