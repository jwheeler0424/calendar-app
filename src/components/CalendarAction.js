import React from 'react';
import { connect } from 'react-redux';
import AddEventPage from './AddEventPage';
import EditEventPage from './EditEventPage';
import ViewEventPage from './ViewEventPage';
import EventListPage from './EventListPage';

export const CalendarAction = (props) => {
    switch (props.views.activeView) {
        case 'list':
            return (<EventListPage />);
        case 'add':
            return (<AddEventPage />);
        case 'edit':
            return (<EditEventPage />);
        case 'view':
            return (<ViewEventPage />);
        default:
            console.log(window.innerWidth)
            if(window.innerWidth >= 1172) {
                return (<EventListPage />)
            } else {
                return (<></>);
            }
    }
}

const mapStateToProps = (state) => ({
    views: state.views
});

export default connect(mapStateToProps)(CalendarAction);