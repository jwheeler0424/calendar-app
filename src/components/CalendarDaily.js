import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import { NavigateBefore, NavigateNext } from '../svg/Icons';
import { setCurrentDate } from '../actions/views';
import CalendarDay from './CalendarDay';

export class CalendarDaily extends React.Component {
    setDayPrev = () => {
        const prevDay = moment(this.props.views.currentDate).subtract(1, 'days').valueOf();
        this.props.setCurrentDate(prevDay);
    }
    setDayNext = () => {
        const nextDay = moment(this.props.views.currentDate).add(1, 'days').valueOf();
        this.props.setCurrentDate(nextDay)
    }
    render () {
        const prevDay = moment(this.props.views.currentDate).subtract(1, 'days');
        const nextDay = moment(this.props.views.currentDate).add(1, 'days');
        return (
            <div className="calendar-daily__wrapper">
                <button onClick={this.setDayPrev}><NavigateBefore /> {prevDay.format('MMM Do')}</button>
                <button onClick={this.setDayNext}>{nextDay.format('MMM Do')} <NavigateNext /></button>
                <h1>{moment(this.props.views.currentDate).format('MMMM Do, YYYY')}</h1>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    views: state.views
});

const mapDispatchToProps = (dispatch) => ({
    setCurrentDate: (currentDate) => dispatch(setCurrentDate(currentDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(CalendarDaily);