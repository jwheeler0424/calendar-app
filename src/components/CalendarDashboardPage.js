import React from 'react';
import { connect } from 'react-redux';
import CalendarSelector from './CalendarSelector';
import CalendarView from './CalendarView';
import CalendarAction from './CalendarAction';

const CalendarDashboardPage = (props) => {
    return (
        <main>
            <CalendarSelector />
            <div className ="calendar-active__wrapper content-container">
                <CalendarView />
                <CalendarAction />
            </div>
        </main>
    );
}

const mapStateToProps = (state) => ({
    views: state.views
});

export default connect(mapStateToProps)(CalendarDashboardPage);