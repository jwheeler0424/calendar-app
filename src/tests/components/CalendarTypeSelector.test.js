import React from 'react';
import { shallow } from 'enzyme';
import { CalendarTypeSelector } from '../../components/CalendarTypeSelector';

test('should render CalendarTypeSelector corrently', () => {
    const wrapper = shallow(<CalendarTypeSelector />);
    expect(wrapper).toMatchSnapshot();
});

test('should set activeCalendar to daily', () => {
    const wrapper = shallow(<CalendarTypeSelector />);
    wrapper.find('button').at(0).simulate('click');
    
});