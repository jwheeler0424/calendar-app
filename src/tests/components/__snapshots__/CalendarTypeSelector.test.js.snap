// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CalendarTypeSelector correctly 1`] = `
<div
  className="calendar-selector__wrapper"
>
  <button
    className="calendar-selector__selector"
    onClick={[MockFunction]}
  >
    Daily
  </button>
  <button
    className="calendar-selector__selector"
    onClick={[MockFunction]}
  >
    Weekly
  </button>
  <button
    className="calendar-selector__active"
    onClick={[MockFunction]}
  >
    Monthly
  </button>
  <button
    className="calendar-selector__selector"
    onClick={[MockFunction]}
  >
    Yearly
  </button>
</div>
`;
